server/project/accounts/models.py

```diff
@@ -52,10 +54,15 @@ def is_debit(self):
     def account_number(self):
         return (self.account_type.liquidity * NUM_ACCOUNTS_PER_ACCOUNT_TYPE) + self.order
 
-    def get_balance(self):
+    def get_balance(self, as_of=None):
```

Comment:
I think this is not doing the right thing. It is summing the transactions then flipping the sign at the end before summing it with the initial balance. Shouldn't it be taking each individual transaction, flipping the sign if it is the opposite charge as the account it is being applied to, then summing everything?

Reply:
Never mind, works fine!

Reply:
Hmm...I think it's still not quite right. But for a different reason this time. The balance shouldn't be flipped just because the account is a debit or credit account. If there is a negative balance, it should be because the sum of transactions is skewed toward the side that is not the normal side? As in, if Cash is negative, then there was a higher amount of credits vs. debits, but vice-versa for an account like Accounts Payable. Sounds right?

Reply:
Discussed and the function is working as expected.